src/runtime/virtcontainers/sandbox.go

```diff
@@ -995,6 +1004,104 @@ func (cw *consoleWatcher) stop() {
 	}
 }
 
+func (s *Sandbox) addSwap(ctx context.Context, id string, size int64) (*config.BlockDrive, error) {
+	swapFile := filepath.Join(getSandboxPath(s.id), id)
+
+	swapFD, err := os.OpenFile(swapFile, os.O_CREATE, 0600)
+	if err != nil {
+		err = fmt.Errorf("creat swapfile %s fail %s", swapFile, err.Error())
+		s.Logger().Error(err)
+		return nil, err
+	}
+	swapFD.Close()
+	defer func() {
+		if err != nil {
+			os.Remove(swapFile)
+		}
+	}()
+
+	err = os.Truncate(swapFile, size)
+	if err != nil {
+		err = fmt.Errorf("truncate swapfile %s fail %s", swapFile, err.Error())
+		s.Logger().Error(err)
```

Comment:
NIT: Log should be `.WithError(err).Error("some info")`

Reply:
I set swapFile and err.Error() to err than they can return back to caller with err.
Do you mind I keep this part?

Reply:
Done.